src/qt/walletcontroller.cpp

```diff
@@ -29,15 +33,17 @@ WalletController::WalletController(interfaces::Node& node, const PlatformStyle*
         getOrCreateWallet(std::move(wallet));
     }
 
-    m_activity_thread.start();
+    m_activity_worker->moveToThread(m_activity_thread);
```

Comment:
Could add
```
    connect(m_activity_thread, &QThread::finished, m_activity_worker, &QObject::deleteLater);
```
and remove following
```
   delete m_activity_worker;
```
?

Reply:
Why? The thread is explicitly stopped (actually the event loop is stopped and the thread finishes) in `~WalletController()` so what better place to also delete the worker? I think explicit teardown is preferable.

Reply:
> Why?

Your approach binds the life of `m_activity_worker` to the `WalletController` instance lifetime.
IMO, as `m_activity_worker` lives in `m_activity_thread`, connecting to the `QThread::finished` signal seems more appropriate.